src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala

```diff
@@ -36,6 +36,52 @@ class ErgoTransactionSpec extends ErgoPropertyTest with ErgoTestConstants {
 
   private val emptyModifierId: ModifierId = bytesToId(Array.fill(32)(0.toByte))
 
+
+  private def updateAnAsset(tx: ErgoTransaction, from: IndexedSeq[ErgoBox], deltaFn: Long => Long) = {
+    val updCandidates = tx.outputCandidates.foldLeft(IndexedSeq[ErgoBoxCandidate]() -> false) { case ((seq, modified), ebc) =>
+      if (modified) {
+        (seq :+ ebc) -> true
+      } else {
+        if (ebc.additionalTokens.nonEmpty && ebc.additionalTokens.exists(t => !java.util.Arrays.equals(t._1, from.head.id))) {
+          (seq :+ modifyAsset(ebc, deltaFn, Digest32 @@ from.head.id)) -> true
+        } else {
+          (seq :+ ebc) -> false
+        }
+      }
+    }._1
+    tx.copy(outputCandidates = updCandidates)
+  }
+
+  private def modifyValue(boxCandidate: ErgoBoxCandidate, delta: Long): ErgoBoxCandidate = {
+    new ErgoBoxCandidate(
+      boxCandidate.value + delta,
+      boxCandidate.ergoTree,
+      boxCandidate.creationHeight,
+      boxCandidate.additionalTokens,
+      boxCandidate.additionalRegisters)
+  }
+
+  private def modifyAsset(boxCandidate: ErgoBoxCandidate,
+                          deltaFn: Long => Long,
+                          idToskip: TokenId): ErgoBoxCandidate = {
+    val assetId = boxCandidate.additionalTokens.find(t => !java.util.Arrays.equals(t._1, idToskip)).get._1
+
+    val tokens = boxCandidate.additionalTokens.map { case (id, amount) =>
+      if (java.util.Arrays.equals(id, assetId)) assetId -> deltaFn(amount) else assetId -> amount
+    }
+
+    new ErgoBoxCandidate(
+      boxCandidate.value,
+      boxCandidate.ergoTree,
+      boxCandidate.creationHeight,
+      tokens,
+      boxCandidate.additionalRegisters)
+  }
+
+  private def checkTx(from: IndexedSeq[ErgoBox], wrongTx: ErgoTransaction): Try[Int] = {
+    wrongTx.statelessValidity().flatMap(_ => wrongTx.statefulValidity(from, emptyDataBoxes, emptyStateContext))
```

Comment:
please add comment on why it is called `wrongTx`?

Reply:
Out of scope of this PR, renamed to tx anyway